mlflow/tracking/fluent.py

```diff
@@ -124,35 +122,30 @@ def start_run(run_uuid=None, experiment_id=None, source_name=None, source_versio
             parent_run_id = None
 
         exp_id_for_run = experiment_id if experiment_id is not None else _get_experiment_id()
-        if is_in_databricks_notebook():
-            databricks_tags = {}
-            notebook_id = get_notebook_id()
-            notebook_path = get_notebook_path()
-            webapp_url = get_webapp_url()
-            if notebook_id is not None:
-                databricks_tags[MLFLOW_DATABRICKS_NOTEBOOK_ID] = notebook_id
-            if notebook_path is not None:
-                databricks_tags[MLFLOW_DATABRICKS_NOTEBOOK_PATH] = notebook_path
-            if webapp_url is not None:
-                databricks_tags[MLFLOW_DATABRICKS_WEBAPP_URL] = webapp_url
-            active_run_obj = MlflowClient().create_run(
-                experiment_id=exp_id_for_run,
-                run_name=run_name,
-                source_name=notebook_path,
-                source_version=source_version or _get_source_version(),
-                entry_point_name=entry_point_name,
-                source_type=SourceType.NOTEBOOK,
-                tags=databricks_tags,
-                parent_run_id=parent_run_id)
-        else:
-            active_run_obj = MlflowClient().create_run(
-                experiment_id=exp_id_for_run,
-                run_name=run_name,
-                source_name=source_name or _get_source_name(),
-                source_version=source_version or _get_source_version(),
-                entry_point_name=entry_point_name,
-                source_type=source_type or _get_source_type(),
-                parent_run_id=parent_run_id)
+
+        user_specified_tags = {}
+        if source_name is not None:
+            user_specified_tags[MLFLOW_SOURCE_NAME] = source_name
+        if source_type is not None:
+            user_specified_tags[MLFLOW_SOURCE_TYPE] = source_type
+        if source_version is not None:
+            user_specified_tags[MLFLOW_GIT_COMMIT] = source_version
+        if entry_point_name is not None:
+            user_specified_tags[MLFLOW_PROJECT_ENTRY_POINT] = entry_point_name
+
+        tags = context.resolve_tags(user_specified_tags)
+
+        active_run_obj = MlflowClient().create_run(
+            experiment_id=exp_id_for_run,
+            run_name=run_name,
```

Comment:
Hmm, a bit weird, I thought run_name was also a tag.

Reply:
It is, but it's set in each of the stores separately - I'd like to bring that into a higher layer in a future PR.